router/product.js

```diff
@@ -0,0 +1,115 @@
+import express from "express";
+import { PrismaClient } from "@prisma/client";
+import { BadRequestError, InternalServerError } from "../middleware/error.js";
+
+const prisma = new PrismaClient();
+const router = express.Router();
+
+// 유효성 검사
+import { validateProduct } from "../middleware/validate.js";
+
+// 상품 등록 + 목록 조회
+router
+  .route("/")
+  .post(validateProduct, async (req, res) => {
+    try {
+      const { name, description, price, tags } = req.body;
+      const product = await prisma.product.create({
+        data: { name, description, price, tags },
+      });
+      res.status(201).json(product);
+    } catch (e) {
+      next(new InternalServerError("상품 등록에 실패했습니다."));
+    }
+  })
+  .get(async (req, res) => {
+    try {
+      const offset = parseInt(req.query.offset) || 0;
+      const limit = parseInt(req.query.limit) || 10;
+      const search = req.query.search?.toString() || "";
+
+      const product = await prisma.product.findMany({
+        where: {
+          OR: [
+            { name: { contains: search } },
+            { description: { contains: search } },
+          ],
+        },
+        skip: offset,
+        take: limit,
+        orderBy: {
+          createdAt: "desc",
+        },
+        select: {
+          id: true,
+          name: true,
+          price: true,
+          createdAt: true,
+        },
+      });
+
+      if (!product)
```

Comment:
빈 배열도 truty한 값이기 때문에 length 체크가 필요합니다.
```suggestion
      if (product.length === 0)
```

Reply:
그리고 보통 리스트 조회의 응답에는 빈 배열이라면 404대신 200에 `[]` 빈 배열을 반환하곤 합니다.